build.ps1

```diff
@@ -19,23 +19,32 @@ param (
     [switch]$Clean,
     # When specified, cargo test will be invoked for the specified component.
     [switch]$Test,
+    # When specified, cargo clippy will be invoked for the specified component.
+    [switch]$Lint,
+    # Run rustfmt
+    [switch]$Fmt,
     # When specified a build will not be run.
     [switch]$SkipBuild,
     # Use a optimized release build
     [switch]$Release,
     # Run a cargo check instead of build
     [switch]$Check,
-    # Add protocol feature
-    [switch]$protocols
+    # Features to pass to cargo
+    [string]$Features,
+    # Options to pass to the cargo test command
+    [string]$TestOptions,
+    # Use nightly toolchain
+    [switch]$Nightly
 )
+$ErrorActionPreference="stop"
+
+$toolchain = "stable"
+if($Nightly) { $toolchain = "nightly" }
+elseif($Fmt) { $toolchain = "nightly-2019-04-09" }
```

Comment:
We should make sure the instructions for bumping the nightly rustfmt during a release point out this line as something to update.

Is there a way we can define this in one place, and then pull from that source in the various build tooling we have lying around?